core/src/main/java/org/apache/iceberg/jdbc/JdbcTableOperations.java

```diff
@@ -182,18 +169,13 @@ private void createTable(String newMetadataLocation) throws SQLException, Interr
           tableIdentifier, catalogName, namespace);
     }
 
+    if (JdbcUtil.viewExists(catalogName, connections, tableIdentifier)) {
```

Comment:
It seems like we should be checking if a table or view exists here.  Either that or we just allow the primary key constraint to fail like I assume it would if the table already exists currently.

Reply:
Same comment as for table/view exists check. I will test that again, but I'm pretty sure the tests won't be happy 😄